terraform/azure-mongodb/azure-mongodb.tf

```diff
@@ -47,9 +49,15 @@ provider "azurerm" {
   skip_provider_registration = var.skip_provider_registration
 }
 
+resource "random_string" "random" {
+  length = 8
+  special = false
+  upper = false
+}
+
 locals {
 	resource_group = length(var.resource_group) == 0 ? format("rg-%s", var.instance_name) : var.resource_group
-
+	private_endpoint_enabled = var.private_endpoint_subnet_id == null ? false : length(var.private_endpoint_subnet_id) > 0 ? true : false
 	enable_virtual_network_filter = (var.authorized_network != "")
 }
 
```

Comment:
I think you should also set `public_network_access_enabled` in the `azurerm_cosmosdb_account` resource to disable public network access when the private endpoint is being used.

```
  public_network_access_enabled = local.private_endpoint_enabled ? false : true
```

Reply:
I've added the proposed changed. Should the public_network_access_enabled value be a plan_input instead to keep the tf template more generic? (not sure if everyone using the broker will want to disable public access if private endpoints are configured)

Reply:
That sounds good. 👍

Reply:
refactored to be parametrized